services/core-web/src/components/noticeOfWork/applications/FinalPermitDocuments.js

```diff
@@ -0,0 +1,235 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Button, Icon, Progress, notification } from "antd";
+import { getDocumentDownloadToken } from "@common/utils/actionlessNetworkCalls";
+import { getDocumentDownloadState } from "@common/selectors/noticeOfWorkSelectors";
+import {
+  setNoticeOfWorkApplicationDocumentDownloadState,
+  fetchImportedNoticeOfWorkApplication,
+  updateNoticeOfWorkApplication,
+} from "@common/actionCreators/noticeOfWorkActionCreator";
+import { openModal, closeModal } from "@common/actions/modalActions";
+import { EDIT_OUTLINE } from "@/constants/assets";
+import { modalConfig } from "@/components/modalContent/config";
+import { COLOR } from "@/constants/styles";
+import CustomPropTypes from "@/customPropTypes";
+import NOWDocuments from "@/components/noticeOfWork/applications//NOWDocuments";
+
+/**
+ * @class FinalPermitDocuments- call logic surrounding adding or removing documents in the final Permit document list
+ */
+
+const propTypes = {
+  mineGuid: PropTypes.string.isRequired,
+  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+  updateNoticeOfWorkApplication: PropTypes.func.isRequired,
+  fetchImportedNoticeOfWorkApplication: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  openModal: PropTypes.func.isRequired,
+  setNoticeOfWorkApplicationDocumentDownloadState: PropTypes.func.isRequired,
+  documentDownloadState: CustomPropTypes.documentDownloadState.isRequired,
+};
+
+export class FinalPermitDocuments extends Component {
+  state = {
+    cancelDownload: false,
+  };
+
+  createFinalDocumentPackage = (selectedCoreRows) => {
+    const documentPayload = this.props.noticeOfWork.documents.map((document) => {
+      // eslint-disable-next-line no-unused-expressions
+      selectedCoreRows.includes(document.now_application_document_xref_guid)
```

Comment:
`document.is_final_package = selectedCoreRows.includes(document.now_application_document_xref_guid);` is a bit easier to read in my opinion, but it's no big deal.

Reply:
ah, yes agree 🤦